src/main/java/org/terasology/gookeeper/system/HungerSystem.java

```diff
@@ -0,0 +1,177 @@
+import org.terasology.gookeeper.component.HungerComponent;
+import org.terasology.gookeeper.event.GooeyFedEvent;
+import org.terasology.logic.characters.CharacterHeldItemComponent;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.common.DisplayNameComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.physics.Physics;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.WorldProvider;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+@Share(value = HungerSystem.class)
+public class HungerSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+
+    @In
+    private WorldProvider worldProvider;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private BlockEntityRegistry blockEntityRegistry;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    @In
+    private InventoryManager inventoryManager;
+
+    @In
+    private Physics physics;
+
+    @In
+    private AssetManager assetManager;
+
+    @In
+    private PrefabManager prefabManager;
+
+    @In
+    private DelayManager delayManager;
+
+    @In
+    private NUIManager nuiManager;
+
+    private static final Logger logger = LoggerFactory.getLogger(HungerSystem.class);
+    private Random random = new FastRandom();
+    private static final String eventID = "DECREASE_HEALTH_TICK";
+
+    @Override
+    public void initialise() {
+        /* Adds health degradation action for the already captured gooey entities. */
+        for (EntityRef gooey : entityManager.getEntitiesWith(GooeyComponent.class)) {
+            GooeyComponent gooeyComponent = gooey.getComponent(GooeyComponent.class);
+            HungerComponent hungerComponent = gooey.getComponent(HungerComponent.class);
+
+            if (gooeyComponent.isCaptured) {
+                delayManager.addPeriodicAction(gooey, eventID, hungerComponent.timeBeforeHungry, hungerComponent.healthDecreaseInterval);
+            }
+        }
+    }
+
+    @Override
+    public void shutdown() {
+    }
+
+    @Override
+    public void update(float delta) {
+
+    }
+
+    /**
+     * Receives ActivateEvent when the targeted gooey is 'activated' and then is fed the held food block.
+     *
+     * @param event,entity   The ActivateEvent, the gooey entity
+     */
+    @ReceiveEvent(components = {GooeyComponent.class})
+    public void onGooeyActivated(ActivateEvent event, EntityRef entity) {
+        if (event.getTarget().hasComponent(GooeyComponent.class)) {
```

Comment:
I think this is a repeat of the logic that only sends this event to entities that have gooeyComponent (from your ```@RecieveEvent``` configuration)